tests/failpoints/cases/test_gc_worker.rs

```diff
@@ -74,3 +77,184 @@ fn test_gcworker_busy() {
+
+    // Pause store-2 after calling observer callbacks and before writing to the rocksdb.
+    let new_leader_apply_fp = "post_handle_apply_1003";
+    fail::cfg(new_leader_apply_fp, "pause").unwrap();
+
+    // Write 1 lock.
+    must_kv_prewrite(
+        &store_1_client,
+        ctx,
+        vec![new_mutation(Op::Put, b"k1", b"v")],
+        b"k1".to_vec(),
+        10,
+    );
+    // Wait for store-2 applying.
+    std::thread::sleep(Duration::from_secs(3));
+
+    // Starting the process of green GC at safe point 20:
+    //   1. Register lock observers on all stores.
+    //   2. Scan locks physically on each store independently.
+    //   3. Get locks from all observers.
+    let safe_point = 20;
+
+    // Register lock observers.
+    clients.iter().for_each(|(_, c)| {
+        must_register_lock_observer(c, safe_point);
+    });
+
+    // Finish scanning locks on store-2 and find nothing.
+    let store_2_client = clients.get(&new_peer.get_store_id()).unwrap();
+    let locks = must_physical_scan_lock(store_2_client, Context::default(), safe_point, b"", 1);
+    assert!(locks.is_empty(), "{:?}", locks);
+
+    // Transfer the region from store-1 to store-2.
+    fail::remove(new_leader_apply_fp);
+    cluster.must_transfer_leader(region_id, new_peer);
+    cluster.pd_client.must_remove_peer(region_id, leader);
+    // Wait for store-1 desroying the region.
+    std::thread::sleep(Duration::from_secs(3));
+
+    // Scan locks on store-1 after the region has been destroyed.
+    let locks = must_physical_scan_lock(store_1_client, Context::default(), safe_point, b"", 1);
+    assert!(locks.is_empty(), "{:?}", locks);
+
+    // Check lock observers.
+    clients.iter().for_each(|(_, c)| {
+        let locks = must_check_lock_observer(c, safe_point, true);
+        assert!(locks.is_empty(), "{:?}", locks);
+    });
+
+    // Re-scan the store-2 and find the lock.
+    let locks = must_physical_scan_lock(&store_2_client, Context::default(), safe_point, b"", 10);
+    assert_eq!(locks.len(), 1);
+    assert_eq!(locks[0].get_key(), b"k1");
+}
+
+#[test]
+fn test_observer_send_error() {
+    let (_cluster, client, ctx) = must_new_cluster_and_kv_client();
+
+    let max_ts = 100;
+    must_register_lock_observer(&client, max_ts);
+    must_kv_prewrite(
+        &client,
+        ctx.clone(),
+        vec![new_mutation(Op::Put, b"k1", b"v")],
+        b"k1".to_vec(),
+        10,
+    );
+    assert_eq!(must_check_lock_observer(&client, max_ts, true).len(), 1);
+
+    let observer_send_fp = "lock_observer_send";
+    fail::cfg(observer_send_fp, "return").unwrap();
+    must_kv_prewrite(
+        &client,
+        ctx.clone(),
+        vec![new_mutation(Op::Put, b"k2", b"v")],
+        b"k1".to_vec(),
+        10,
+    );
+    let resp = check_lock_observer(&client, max_ts);
+    assert!(resp.get_error().is_empty(), "{:?}", resp.get_error());
+    // Should mark clean if fails to send locks.
+    assert!(!resp.get_is_clean());
```

Comment:
I'm afraid there may be some issue when checking happens between send error occurs and marking dirty.

Reply:
IMO, it's not an issue.